boards/xtensa/m5stack_core2/Kconfig.board

```diff
@@ -0,0 +1,8 @@
+# M5Stack Core2 board configuration
+
+# Copyright (c) 2023 Martin Kiepfer <m.kiepfer@teleschirm.org>
+# SPDX-License-Identifier: Apache-2.0
+
```

Comment:
Can you check whether it is missing `SOC_PART_NUMBER`? This would define PSRAM automatically:

```c
choice SOC_PART_NUMBER
	default SOC_ESP32_WROVER_E_N16R8
endchoice
```

Reply:
M5Stack-Core2 is featuring SOC_ESP32_D0WD_V3. This is very similar to SOC_ESP32_WROVER_E_N16R8, but I find it more consistent to keep sram and flash definition separately within board layer.